docs/content/developer/cryptography/on-chain/groth16.mdx

```diff
@@ -83,37 +112,53 @@ fn main() {
     let proof = Groth16::<Bn254>::prove(&params, circom, &mut rng).unwrap();
 
     // Check that the proof is valid
-    let pvk = Groth16::<Bn254>::process_vk(&params.vk).unwrap();
+    let pvk = prepare_verifying_key(&params.vk);
     let verified = Groth16::<Bn254>::verify_with_processed_vk(&pvk, &inputs, &proof).unwrap();
     assert!(verified);
 }
 ```
 The proof shows that an input (7) which, when hashed with the Poseidon hash function, gives a certain output (which in this case is `inputs[0].to_string() = 7061949393491957813657776856458368574501817871421526214197139795307327923534`).
 
+You can also use snarkjs to generate and export proof yourself to observe the proof body.
+```bash
+snarkjs groth16 prove circuit_0001.zkey witness.wtns proof.json public.json
+```
+
 ### Verification in IOTA
 
-The API in IOTA for verifying a proof expects a special processed verification key, where only a subset of the values are used. Ideally, computation for this prepared verification key happens only once per circuit. You can perform this processing using the `iota::groth16::prepare_verifying_key` method of the IOTA Move API with a serialization of the `params.vk` value used previously.
+The API in IOTA for verifying a proof expects a special processed verification key, where only a subset of the values are used. Ideally, computation for this prepared verification key happens only once per circuit. You can perform this processing using the `iota::groth16::prepare_verifying_key` method of the IOTA Move API with a serialization of the `params.vk` value used previously. Remember that you should not use this in a production environment, as you should grab a safer entropy resource i.e., from a ceremony.
```

Comment:
> Remember that you should not use this in a production environment, as you should grab a safer entropy resource i.e., from a ceremony.

Weird phrasing. What exactly "this" should not be used in production? Verification key processing with `iota::groth16::prepare_verifying_key`? Why entropy "resource" here? The phrase doesn't make sense to me, it would be even more confusing to users.